boards/sparkfun/pro_micro_rp2040/sparkfun_pro_micro_connector.dtsi

```diff
@@ -11,8 +11,8 @@
 		gpio-map-mask = <0xffffffff 0xffffffc0>;
 		gpio-map-pass-thru = <0 0x3f>;
 		gpio-map
-			= <0 0 &gpio0 0 0>		/* D0 */
-			, <1 0 &gpio0 1 0>		/* D1 */
+			= <1 0 &gpio0 0 0>		/* D1 */
+			, <0 0 &gpio0 1 0>		/* D0 */
```

Comment:
trying to wrap my head around to problem, say that someone made a shield that uses the nexus node and works for a pro micro, this would make it work seamlessly with the rp2040 right?

Reply:
Exactly right. We use this extensively for ZMK, and we added this same fix to our Zephyr fork recently to address this bug.

Reply:
Got it, so I think the change should go in but there should be some comment highlighting the fact that there are, sadly, two variants, and some of these looks off because of that.

Reply:
I think the only "SparkFun Pro Micro" variant supported by Zephyr is the RP2040-based one, since the original Pro Micro is based on ATmega32u4, which is not supported as far as I understand? So there will be no conflicting versions. Or is there yet another variant?

Like Pete mentioned, many shields (in ZMK) are designed to be "Pro Micro" compatible, I suppose primarily for historical reasons.

Just for clarity: Is me that should add a note to the 3.7 migration guide? Or will it be handled elsewhere?

Reply:
> Just for clarity: Is it me that should add a note to the 3.7 migration guide? Or will it be handled elsewhere?

Yes please, migration guide changes should go with the PR doing the change itself. Just a sentence indicating what to do and when, nothing fancy.

Reply:
I finally got around to figure out where to update what, and added a short instruction/description in the migration guide. That should do it, then?